### PR TITLE
add e2e testcase fuse-recovery

### PR DESCRIPTION
**Pull request by baowj-678**
Signed-off-by: baowj-678 <bwj_678@qq.com>

### Ⅰ. Describe what this PR does

add e2e testcase fuse-recovery([6-Fuse挂载点自动恢复](https://github.com/fluid-cloudnative/community/blob/master/testcase/fluid-testcase.md#6-fuse%E6%8C%82%E8%BD%BD%E7%82%B9%E8%87%AA%E5%8A%A8%E6%81%A2%E5%A4%8D))

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: test/prow/fuse_recovery.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+                continue
+        print("PersistentVolume & PersistentVolumeClaim Ready.")
+        break
+
+
+def enableInjection():
+    api = client.CoreV1Api()
+    body = {'metadata': {
+        'labels': {
+            'fluid.io/enable-injection': 'true'
```
Comment (TrafalgarZZZ) [inline test/prow/fuse_recovery.py:152]: In the latest  Fluid version, no need to add `fluid.io/enable-injection=true` label on namespace.

File: test/prow/fuse_recovery.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    """
+    fluid_namespace = "fluid-system"
+    pod_name = "csi-nodeplugin-fluid"
+    pod_name = getPodNameByPrefix(pod_name, fluid_namespace)
+    if pod_name is None:
+        return False
+    api = client.CoreV1Api()
+    while True:
+        log_content = api.read_namespaced_pod_log(pod_name, fluid_namespace, container="plugins")
+        if str(log_content).__contains__("start csi recover"):
```
[the remark this answers is not in the document]
Comment (baowj-678) [reply test/prow/fuse_recovery.py:47]: yes, container log may be rollover, I'll check pod spec.

File: test/prow/fuse_recovery.py (hunk 1 above)
Comment (baowj-678) [reply test/prow/fuse_recovery.py:152]: I referred to some old docs, I'll delete it.

**Comment by TrafalgarZZZ** [comment]
/hold
hold because fuse recovery now may not work due to unknown reason

File: test/prow/fuse_recovery.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        time.sleep(1)
+
+
+def checkFuseRecovered():
+    api = client.CoreV1Api()
+    while True:
+        items = api.list_namespaced_event(namespace=namespace).items
+        for item in items:
+            if item.message.__contains__("Fuse recover"):
+                print("Fuse Recovered.")
```
Comment (TrafalgarZZZ) [inline test/prow/fuse_recovery.py:126]: Event is a kind of K8s resource will exist for a long time. So consider the following steps:
1. you create a dataset
2. you delete a fuse pod to trigger fuse recovery
3. Fluid successfully recovered a FUSE mountpoint from dataset. That is, Fluid sends an event containing `Fuse recover`
4. you delete the dataset
5. you recreate a dataset with the same name
6. again, you delete a fuse pod to trigger fuse recovery
7. **At this time, the code above may get phantom events from the last deleted dataset, so the `if` statement returns true and goes on. but the new dataset has not been recovered.**

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]

**Comment by TrafalgarZZZ** [comment]
@baowj-678 Could you please rebase the latest master branch? CSI Recover has already been fixed in #2747

**Comment by baowj-678** [comment]
> @baowj-678 Could you please rebase the latest master branch? CSI Recover has already been fixed in #2747 . We can now get this PR a retest

I have rebased this branch on the latest master branch

**Review by TrafalgarZZZ — APPROVED** [review]
/lgtm

Much thanks to @baowj-678 . Great work for contributing the testcase

**Pull request merged**